pkg/frontend/transport/handler.go

```diff
@@ -489,30 +491,38 @@ func (f *Handler) reportQueryStats(r *http.Request, source, userID string, query
 	}
 
 	var reason string
-	if statusCode == http.StatusTooManyRequests {
-		reason = reasonTooManyRequests
-	} else if statusCode == http.StatusRequestEntityTooLarge {
+	if statusCode == http.StatusRequestEntityTooLarge {
 		reason = reasonResponseBodySizeExceeded
-	} else if statusCode == http.StatusUnprocessableEntity {
+	} else if error != nil {
+		// We are unable to use errors.As to compare since body string from the http response is wrapped as an error
 		errMsg := error.Error()
-		if strings.Contains(errMsg, limitTooManySamples) {
-			reason = reasonTooManySamples
-		} else if strings.Contains(errMsg, limitTimeRangeExceeded) {
-			reason = reasonTimeRangeExceeded
-		} else if strings.Contains(errMsg, limitSeriesFetched) {
-			reason = reasonSeriesFetched
-		} else if strings.Contains(errMsg, limitChunksFetched) {
-			reason = reasonChunksFetched
-		} else if strings.Contains(errMsg, limitChunkBytesFetched) {
-			reason = reasonChunkBytesFetched
-		} else if strings.Contains(errMsg, limitDataBytesFetched) {
-			reason = reasonDataBytesFetched
-		} else if strings.Contains(errMsg, limitSeriesStoreGateway) {
-			reason = reasonSeriesLimitStoreGateway
-		} else if strings.Contains(errMsg, limitChunksStoreGateway) {
-			reason = reasonChunksLimitStoreGateway
-		} else if strings.Contains(errMsg, limitBytesStoreGateway) {
-			reason = reasonBytesLimitStoreGateway
+		if statusCode == http.StatusTooManyRequests {
+			var resourceExhaustedErr resource.ExhaustedError
+			if strings.Contains(errMsg, resourceExhaustedErr.Error()) {
```

Comment:
Will this also capture the token byte limiter error https://github.com/cortexproject/cortex/blob/531dd58247bb6bb2da3c912e9d990a0a4482f833/pkg/storegateway/limiter.go#L18?

Reply:
oops, thanks for catching that